src/titiler/xarray/titiler/xarray/io.py

```diff
@@ -138,27 +138,37 @@ def _arrange_dims(da: xarray.DataArray) -> xarray.DataArray:
 def get_variable(
     ds: xarray.Dataset,
     variable: str,
-    datetime: Optional[str] = None,
-    drop_dim: Optional[List[str]] = None,
+    sel: Optional[List[str]] = None,
+    method: Optional[Literal["nearest", "pad", "ffill", "backfill", "bfill"]] = None,
 ) -> xarray.DataArray:
     """Get Xarray variable as DataArray.
 
     Args:
         ds (xarray.Dataset): Xarray Dataset.
         variable (str): Variable to extract from the Dataset.
-        datetime (str, optional): datetime to select from the DataArray.
-        drop_dim (list of str, optional): DataArray dimension to drop in form of `[{dimension}={value}],`.
+        sel (list of str, optional): List of Xarray Indexes.
+        method (str): Xarray indexing method.
 
     Returns:
         xarray.DataArray: 2D or 3D DataArray.
 
     """
     da = ds[variable]
 
-    if drop_dim:
-        for dim in drop_dim:
-            dim_to_drop, dim_val = dim.split("=")
-            da = da.sel({dim_to_drop: dim_val}).drop_vars(dim_to_drop)
+    if sel:
+        sel_idx = []
+        for s in sel:
+            val: Union[str, slice]
+            dim, val = s.split("=")
+            dt = da[dim].dtype
+            if "," in val:
+                val = slice(*[numpy.array(v).astype(dt) for v in val.split(",")])
```

Comment:
I took a chance here, would a value have `,`? 🤷 

basically this is to support things like `time=date1,date2` -> da.sel(time=slice(date1, date2))`

Reply:
You could branch on the dimension dtype being string and in that case not do the slice. 🤷 

Otherwise, I cannot think of a case where a value would contain comma. So perhaps just leave as-is until someone hits this issue.

Reply:
😓 I guess we need to come up with something else.

Reply:
Nah, I think it's fair to expect that people do not use the most [obscure](https://xkcd.com/1179/) ways of specifying datetime.

So I'd stick to the splitting on comma - just raise good errors when it fails.

Reply:
we could also use `slice` is user pass twice the same dim `time=date1,time=date2` -> `time=slice(date1, date2)` 🤷‍♂️

Reply:
Wait, why do we do slices at all? Apart from maybe the spatial dimensions, you always need to reduce the dimensions to 1, i.e. select a single index, no?

Passing the same dim twice would, to me, mean `sel(time=[date1, date2])` - but same thing here re reducing dimensionality.

Or are you planning to treat multiple 2D fields as "bands"?

Reply:
> Or are you planning to treat multiple 2D fields as "bands"?

Yes we do support 3D data now, meaning you could have multiple Z (time or dim) represented as bands